common/db/client_credentials.go

```diff
@@ -0,0 +1,234 @@
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+// GetClientCredential retrieves a client credential by client_id
+func (m *MongoStore) GetClientCredential(clientID string) (models.ClientCredentials, error) {
+	coll := m.GetCollection("client_credentials")
+	filter := bson.D{{Key: "client_id", Value: clientID}}
+	return GetDocument[models.ClientCredentials](m.Ctx, coll, filter)
+}
+
+// GetClientCredentialsByUser retrieves all client credentials for a specific user
+func (m *MongoStore) GetClientCredentialsByUser(userID string) ([]models.ClientCredentials, error) {
+	coll := m.GetCollection("client_credentials")
+	filter := bson.D{{Key: "user_id", Value: userID}, {Key: "is_active", Value: true}}
+	return GetDocuments[models.ClientCredentials](m.Ctx, coll, filter)
+}
+
+// GetClientCredentialsByTenant retrieves all client credentials for a specific tenant
+func (m *MongoStore) GetClientCredentialsByTenant(tenant string) ([]models.ClientCredentials, error) {
+	coll := m.GetCollection("client_credentials")
+	filter := bson.D{{Key: "tenant", Value: tenant}, {Key: "is_active", Value: true}}
+	return GetDocuments[models.ClientCredentials](m.Ctx, coll, filter)
+}
+
+// CreateClientCredential creates a new client credential with generated client_id and secret
+func (m *MongoStore) CreateClientCredential(name, description, userID, tenant string) (models.ClientCredentials, error) {
+	clientID := clientcredentials.GenerateClientIDWithPrefix(getClientIDPrefix())
+
+	clientSecret, err := clientcredentials.GenerateClientSecret()
+	if err != nil {
+		return models.ClientCredentials{}, fmt.Errorf("failed to generate client secret: %w", err)
+	}
+
+	hashedSecret, err := clientcredentials.HashSecret(clientSecret)
+	if err != nil {
+		return models.ClientCredentials{}, fmt.Errorf("failed to hash client secret: %w", err)
+	}
+
+	credential := models.ClientCredentials{
+		ClientID:     clientID,
+		ClientSecret: hashedSecret,                           // Store hashed version
+		SecretLookup: generateSecretLookupHash(clientSecret), // Store SHA256 for lookup
+		Name:         name,
+		Description:  description,
+		UserID:       userID,
+		Tenant:       tenant,
+		IsActive:     true,
+		CreatedAt:    time.Now(),
+		UpdatedAt:    time.Now(),
+	}
+
+	coll := m.GetCollection("client_credentials")
+	result, err := InsertDocumentWithResult(m.Ctx, coll, credential)
+	if err != nil {
+		return models.ClientCredentials{}, fmt.Errorf("failed to create client credential: %w", err)
+	}
+
+	if oid, ok := result.InsertedID.(primitive.ObjectID); ok {
+		credential.ID = oid.Hex()
+	}
+
+	// Return credential with unhashed secret for client to save
+	credential.ClientSecret = clientSecret
+	return credential, nil
+}
+
+// UpdateClientCredential updates an existing client credential
+func (m *MongoStore) UpdateClientCredential(clientID string, updates map[string]interface{}) error {
```

Comment:
I assume `map[string]interface{}` works as well but shouldn't we be using `bson.M` instead to keep convention for the Mongo client?